File: authentication-service/utils/jwt_utils.go
package utils

import (
	"os"
	"log"
	"time"

	"authentication-service/models"
	"github.com/golang-jwt/jwt"
	"github.com/joho/godotenv"
)

func GenerateToken(user models.UserModel) (string, error) {
	err := godotenv.Load(".env.dev")
    if err != nil {
        log.Fatalf("Error loading .env file")
    }
	var jwtSecret = []byte(os.Getenv("JWT_SECRET_KEY"))
	claims := jwt.MapClaims{
		"email": user.Email,
		"name":  user.Name,
		"exp":   time.Now().Add(time.Hour * 24).Unix(), // Token valid for 24 hours
	}

	// Create the token
	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)

	// Sign the token with the secret key
	return token.SignedString(jwtSecret)
}

File: backend/user-service/src/auth/auth_utils/jwtUtils.ts
import jwt from 'jsonwebtoken';
import { config } from '../../config/envConfig'; // centralized config
import { IUser } from '../../models/user';

export const generateToken = (user: IUser): string => {
    return jwt.sign(
        { userId: user._id, email: user.email },
        config.jwtSecret,
        { expiresIn: config.jwtExpiresIn }
    );
};

export const verifyToken = (token: string): any => {
    try {
        return jwt.verify(token, config.jwtSecret);
    } catch (error) {
        throw new Error('Invalid token');
    }
};